lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -358,17 +358,20 @@ GlobalActorAttributeRequest::evaluate(
 
       // ... and not if it's the instance storage of a struct
       if (!var->isStatic() && var->isOrdinaryStoredProperty()) {
-        if (auto *nominal = var->getDeclContext()->getSelfNominalTypeDecl()) {
-          if (isa<StructDecl>(nominal) && !isWrappedValueOfPropWrapper(var)) {
-
-            var->diagnose(diag::global_actor_on_storage_of_value_type,
-                          var->getName(), nominal->getDescriptiveKind())
-              .highlight(globalActorAttr->getRangeWithAt())
-              .warnUntilSwiftVersion(6);
-
-            // In Swift 6, once the diag above is an error, it is disallowed.
-            if (var->getASTContext().isSwiftVersionAtLeast(6))
-              return None;
+          // skip check if from serialized module
+        if (var->getDeclContext()->getParentSourceFile() != nullptr) {
```

Comment:
Would it be better to put this check on line 317, right after "// Collect the attributes"?

Reply:
Hm, I suppose we could just return the `result` after line 320 if it's a decl from a serialized module, because we know it went through typechecking already.

Reply:
Yeah, I think that's a reasonable approach.